grpc/src/main/java/com/linecorp/armeria/common/grpc/GrpcWebUtil.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.grpc;
+
+import static java.util.Objects.requireNonNull;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.client.retry.RetryRuleWithContent;
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+import com.linecorp.armeria.common.grpc.protocol.GrpcHeaderNames;
+
+import io.grpc.ClientInterceptor;
+import io.netty.util.AttributeKey;
+
+/**
+ * Utilities for working with <a href="https://grpc.io/docs/languages/web/basics/">gRPC-Web</a>.
+ *
+ * <p>Note that this class will be removed once a retry {@link ClientInterceptor} is added.
+ * See: https://github.com/line/armeria/issues/2860
```

Comment:
I guess this will still be useful for those who want to access the trailers while implementing something at HTTP-level?